### PR TITLE
Lab2

### PR DESCRIPTION
**Pull request by sevenzing**

### CODE REVIEW

**Comment by Feyd-Rauth** [comment]
LGTM

**Review by Feyd-Rauth — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: '3'
+services:
+  app_python:
+    image: lymarenkolev/currenttime:latest
+    build: app_python
+    ports:
+      - 4444:4444
+  app_golang:
```
Comment (bragov4ik) [inline docker-compose.yml]:
```suggestion
  app_rust:
```

**Review by bragov4ik — CHANGES_REQUESTED** [review]

File: app_python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+RUN python -m poetry config virtualenvs.create false \
+  && python -m poetry install $(test "$PROD" == true && echo "--no-dev") --no-interaction --no-ansi \
+  && python -m poetry shell
+
+COPY . /app
+
+EXPOSE 4444
+ENTRYPOINT ["gunicorn", "app:app", "-b", "0.0.0.0:4444"]
```
Comment (Ilya-Kolomin) [inline app_python/Dockerfile]: Consider not running the app as `root` user.

File: app_rust/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+RUN cargo init
+COPY ./Cargo.toml ./
+RUN cargo build --release \
+    && rm -rf ./src
+
+# build
+COPY ./ ./
+RUN cargo build --release
+
+FROM ubuntu:20.04 as run
```
Comment (Ilya-Kolomin) [inline app_rust/Dockerfile]: 1. Better to specify more concrete version
2. Ubuntu image is not that light-weight for running the production

**Review by Ilya-Kolomin — APPROVED** [review]
Review OK👍

**Review by bragov4ik — APPROVED** [review]

**Review by pminina01 — APPROVED** [review]

**Pull request closed without merging**